src/ZkFold/Base/Protocol/Plonkup/Internal.hs

```diff
@@ -3,227 +3,47 @@
-    show (PlonkSetupParamsProve omega' k1' k2' gs' h0' h1' iPub') =
-        "Setup Parameters (Prove): "
-        ++ show omega' ++ " "
-        ++ show k1' ++ " "
-        ++ show k2' ++ " "
-        ++ show gs' ++ " "
-        ++ show h0' ++ " "
-        ++ show h1' ++ " "
-        ++ show iPub'
-
-data PlonkSetupParamsVerify c1 c2 = PlonkSetupParamsVerify {
-        omega'' :: ScalarField c1,
-        k1''    :: ScalarField c1,
-        k2''    :: ScalarField c1,
-        x2''    :: Point c2,
-        pow''   :: Integer,
-        n''     :: Integer
-    }
-instance (Show (ScalarField c1), Show (BaseField c1), Show (BaseField c2),
-        EllipticCurve c1, EllipticCurve c2) => Show (PlonkSetupParamsVerify c1 c2) where
-    show (PlonkSetupParamsVerify omega'' k1'' k2'' x2'' pow'' n'') =
-        "Setup Parameters (Verify): "
-        ++ show omega'' ++ " "
-        ++ show k1''  ++ " "
-        ++ show k2''  ++ " "
-        ++ show x2''  ++ " "
-        ++ show pow'' ++ " "
-        ++ show n''
-
-data PlonkPermutation n c = PlonkPermutation {
-        s1 :: PolyVec (ScalarField c) n,
-        s2 :: PolyVec (ScalarField c) n,
-        s3 :: PolyVec (ScalarField c) n
-    }
-instance Show (ScalarField c) => Show (PlonkPermutation n c) where
-    show (PlonkPermutation s1 s2 s3) = "Permutation: " ++ show s1 ++ " " ++ show s2 ++ " " ++ show s3
-
-data PlonkCircuitPolynomials n c = PlonkCircuitPolynomials {
-        ql     :: PlonkPolyExtended n c,
-        qr     :: PlonkPolyExtended n c,
-        qo     :: PlonkPolyExtended n c,
-        qm     :: PlonkPolyExtended n c,
-        qc     :: PlonkPolyExtended n c,
-        sigma1 :: PlonkPolyExtended n c,
-        sigma2 :: PlonkPolyExtended n c,
-        sigma3 :: PlonkPolyExtended n c
-    }
-instance Show (ScalarField c) => Show (PlonkCircuitPolynomials n c) where
-    show (PlonkCircuitPolynomials ql qr qo qm qc sigma1 sigma2 sigma3) =
-        "Circuit Polynomials: "
-        ++ show ql ++ " "
-        ++ show qr ++ " "
-        ++ show qo ++ " "
-        ++ show qm ++ " "
-        ++ show qc ++ " "
-        ++ show sigma1 ++ " "
-        ++ show sigma2 ++ " "
-        ++ show sigma3
-
-data PlonkCircuitCommitments c = PlonkCircuitCommitments {
-        cmQl :: Point c,
-        cmQr :: Point c,
-        cmQo :: Point c,
-        cmQm :: Point c,
-        cmQc :: Point c,
-        cmS1 :: Point c,
-        cmS2 :: Point c,
-        cmS3 :: Point c
+{-
+    NOTE: we need to parametrize the type of transcripts because we use BuiltinByteString on-chain and ByteString off-chain.
+    Additionally, we don't want this library to depend on Cardano libraries.
+-}
+
+data Plonkup (i :: Natural) (n :: Natural) (l :: Natural) curve1 curve2 transcript = Plonkup {
+        omega :: ScalarField curve1,
+        k1    :: ScalarField curve1,
+        k2    :: ScalarField curve1,
+        xPub  :: Vector l (Var (Vector i)),
+        ac    :: ArithmeticCircuit (ScalarField curve1) (Vector i) Par1,
```

Comment:
Consider changing the output type to `Vector l` to incorporate public variables as the `acOutput` part of a circuit instead of a separate field in `Plonkup` datatype